atlp-rwanda/cabal-bn-be
# #181339525 On successful registration, users should receive a JWT

### What does this PR do?

- This PR adds user registration implementation with passport.js and JWT

### Description of Task to be completed?

- user register with email and password
- user credentials are hashed and stored in the database
- user receives a token after registration
- use passport.js in the registration
- also validate the user inputs

### How should this be manually tested?

- clone this [repo](https://github.com/atlp-rwanda/cabal-bn-be/) and checkout to [this](https://github.com/atlp-rwanda/cabal-bn-be/tree/feature/%23181339525-user-registration-with-jwt) branch

- run yarn install and also add the .env file copied from .env.sample
- run yarn run dbmigrate:all to instantiate the DB
- open postman use this URL(http://localhost:3000/api/v1/user/register)
- enter an email and password
- you can also enter the first_name, last_name, occupation, or bio
- refer to the picture at the bottom for all the other fields

- it should be in JSON format

- for testing run yarn run test


### What are the relevant pivotal tracker stories?

[#23181339525](https://www.pivotaltracker.com/n/projects/2555280)

### Screenshots (if appropriate)

![Screenshot (14)](https://user-images.githubusercontent.com/83913153/156553286-4d4d3b7f-ccd0-4512-8a1f-02c2cbdeca4a.png)

![Screenshot (17)](https://user-images.githubusercontent.com/83913153/156791009-c95a40de-55ce-401d-8235-d9fd2b0bc1da.png)

![Screenshot (18)](https://user-images.githubusercontent.com/83913153/156791686-4a977fc6-0559-4dc3-8589-63defde6f69b.png)
